scripts/compile_tests/failures_histogram.py

```diff
@@ -0,0 +1,94 @@
+import argparse
+import re
+
+from common import get_testcases, open_test_results, skipped_test
+
+
+"""
+python failures_histogram.py dynamo_logs_for_py311/
+
+Analyzes skip reasons for Dynamo tests and prints a histogram with repro
+commands. You'll need to download the test reports for the Dynamo shards
+and put them under the specified directory.
+"""
+
+
+def skip_reason(testcase):
+    for child in testcase.iter():
+        if child.tag != "skipped":
+            continue
+        return child.attrib["message"]
+    raise AssertionError("no message?")
+
+
+IGNORED_REASONS = {
+    "Policy: we don't run OpInfo tests w/ Dynamo",
+    "Policy: we don't run ModuleInfo tests w/ Dynamo",
+    "NP_VER: fail on CI",
+    "Excluded from CUDA tests",
+    "CUDA not found",
+    "Only runs on cuda",
+    "NumPy not found",
+    "test is slow; run with PYTORCH_TEST_WITH_SLOW to enable test",
+    "Skipped",
+    "Skipped!",
+    "",
+    "test doesn't currently work with dynamo",
```

Comment:
this seems like a good reason to fix?

Reply:
I should probably just delete most things from this list.